go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -770,6 +772,18 @@ func (app *registryApplication) registerRuntime( // nolint: gocyclo
 			)
 			return nil, err
 		}
+
+		// If owner was updated remove stake claim from previous owner.
```

Comment:
Actually, this should also happen when changing the governance model to runtime as the `acctAddr` changed? I guess this transition is never tested in E2E tests as it should otherwise cause the sanity check to fail? Also why isn't the above just using `rt.StakingAddress()` to compute `acctAddr`?

Reply:
> Actually, this should also happen when changing the governance model to runtime as the acctAddr changed

Added

> Also why isn't the above just using rt.StakingAddress() to compute acctAddr

Refactored this whole section to make it cleaner.

> I guess this transition is never tested in E2E tests

Yeah never tested. I'll look if there's an easy way to add it.

Reply:
Added a test for this to `runtime-dynamic` E2E test.